HackIllinois/Misc/Constants.swift

```diff
@@ -11,21 +11,65 @@
 //
 
 import Foundation
+import CoreData
+import HIAPI
 
 // MARK: - Constants
-struct HIConstants {
+class HIConstants {
+
+    static var shared = HIConstants()
+
+    func updateTimes() {
+
+        // Update the times of event
+        TimeService.getTimes()
+            .onCompletion { result in
+                do {
+                    let (times, _) = try result.get()
+                    let timesDict = times.data
+                    guard let eventStartTime = timesDict["eventStart"],
+                        let eventEndTime = timesDict["eventEnd"],
+                        let hackStartTime = timesDict["hackStart"],
+                        let hackEndTime = timesDict["hackEnd"],
+                        let fridayStartTime = timesDict["fridayStart"],
+                        let fridayEndTime = timesDict["fridayEnd"],
+                        let satStartTime = timesDict["saturdayStart"],
+                        let satEndTime = timesDict["saturdayEnd"],
+                        let sunStartTime = timesDict["sundayStart"],
+                        let sunEndTime = timesDict["sundayEnd"]
+                    else {
+                        print("Failed to update times")
+                        return
+                    }
+                    HIConstants.EVENT_START_TIME = eventStartTime
```

Comment:
My recommendation here is to have a model called HIEventTimes or something where the real times are used, and you can keep the constants as hardcoded below for fallback if the times don't come through (and switch to using the HIEventTImes)

Reply:
Made HITimeDataSource